src/engine/controls/bpmcontrol.cpp

```diff
@@ -274,44 +354,68 @@ void BpmControl::adjustBeatsBpm(double deltaBpm) {
         return;
     }
 
-    const mixxx::Bpm bpm = pBeats->getBpmInRange(
-            mixxx::audio::kStartFramePos, frameInfo().trackEndPosition);
-    // FIXME: calling bpm.value() without checking bpm.isValid()
-    const auto centerBpm = mixxx::Bpm(math_max(kBpmAdjustMin, bpm.value() + deltaBpm));
-    mixxx::Bpm adjustedBpm = BeatUtils::roundBpmWithinRange(
-            centerBpm - kBpmAdjustStep / 2, centerBpm, centerBpm + kBpmAdjustStep / 2);
-    const auto newBeats = pBeats->trySetBpm(adjustedBpm);
-    if (!newBeats) {
-        return;
+    activateActionRepeater([this, v]() {
+        slotAdjustBeatsFaster(v);
+    });
+
+    const auto adjustedBeats =
+            pBeats->tryAdjustTempo(frameInfo().currentPosition,
+                    v > 1 ? mixxx::Beats::TempoAdjustment::MuchFaster
+                          : mixxx::Beats::TempoAdjustment::Faster);
+    if (adjustedBeats) {
+        pTrack->trySetBeats(*adjustedBeats);
     }
```

Comment:
What you are suggesting make sense.
I looked into it in the span of that change is wide, since this `std::optional` is already used in the Beat API. I will add an issue to capture your thought and allow us to fix that going forward.

Reply:
#13806

Reply:
It is OK for me to only alter the code in this PR. Changing the other is also welcome of cause. 
I consider this a merge blocker.